pkg/proxy/filter_endpoints.go

```diff
@@ -0,0 +1,204 @@
+/*
```

Comment:
Is this a direct copy from `pkg/proxy/topology.go` or were there any changes as well?

Reply:
Oh, I see now that FilterEndpoints returns separate endpoints for internal/external

Reply:
Do you think we can do the file name change in a separate PR so we can keep the diff on the changes only?

Reply:
Same comment for `pkg/proxy/filter_endpoints_test.go`

Reply:
The commit "`pkg/proxy: rename topology.go to filter_endpoints.go`" is a straight rename, and then later commits make changes.

Reply:
Ah my bad, I missed the separate commit for this change